src/rpc/rawtransaction.cpp

```diff
@@ -258,32 +258,42 @@ UniValue verifytxoutproof(const JSONRPCRequest& request)
     if (request.fHelp || request.params.size() != 1)
         throw std::runtime_error(
             "verifytxoutproof \"proof\"\n"
-            "\nVerifies that a proof points to a transaction in a block, returning the transaction it commits to\n"
-            "and throwing an RPC error if the block is not in our best chain\n"
+            "\nVerifies that a proof points to a transaction in a block, returning the block hash and transactions it commits to,\n"
+            "throwing an RPC error if the block is not in our best chain\n"
             "\nArguments:\n"
             "1. \"proof\"    (string, required) The hex-encoded proof generated by gettxoutproof\n"
-            "\nResult:\n"
-            "[\"txid\"]      (array, strings) The txid(s) which the proof commits to, or empty array if the proof is invalid\n"
+            "\nResult: (empty if proof is invalid)\n"
+            "{\n"
+            "    \"blockhash\"   (string) The blockhash the included proof commits to\n"
+            "    [\"txid\"]      (array, strings) The txid(s) which the proof commits to\n"
+            "}\n"
         );
 
     CDataStream ssMB(ParseHexV(request.params[0], "proof"), SER_NETWORK, PROTOCOL_VERSION | SERIALIZE_TRANSACTION_NO_WITNESS);
     CMerkleBlock merkleBlock;
     ssMB >> merkleBlock;
 
-    UniValue res(UniValue::VARR);
+    UniValue res(UniValue::VOBJ);
+    UniValue txids(UniValue::VARR);
 
     std::vector<uint256> vMatch;
     std::vector<unsigned int> vIndex;
-    if (merkleBlock.txn.ExtractMatches(vMatch, vIndex) != merkleBlock.header.hashMerkleRoot)
+    if (merkleBlock.txn.ExtractMatches(vMatch, vIndex) != merkleBlock.header.hashMerkleRoot) {
         return res;
+    }
 
     LOCK(cs_main);
 
-    if (!mapBlockIndex.count(merkleBlock.header.GetHash()) || !chainActive.Contains(mapBlockIndex[merkleBlock.header.GetHash()]))
+    if (!mapBlockIndex.count(merkleBlock.header.GetHash()) || !chainActive.Contains(mapBlockIndex[merkleBlock.header.GetHash()])) {
         throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Block not found in chain");
+    }
+
+    for (const uint256& hash : vMatch) {
+        txids.push_back(hash.GetHex());
+    }
 
-    for (const uint256& hash : vMatch)
-        res.push_back(hash.GetHex());
+    res.push_back(Pair("txids", txids));
+    res.push_back(Pair("blockhash", merkleBlock.header.GetHash().GetHex()));
```

Comment:
Push `blockhash` first.

Reply:
I'm curious about the reason for this request. Alphabetical ordering?

Reply:
Also matches the documentation above.